### PR TITLE
NEW function in modulebuilder.lib for rewriting all permissions

### PR DESCRIPTION
**Pull request by lamrani002**
# NEW|New function for rewriting all permissions after every update or delete or add

### CODE REVIEW

File: htdocs/core/lib/modulebuilder.lib.php [hunk 1]
```diff
@@ -416,3 +416,70 @@ function rebuildObjectSql($destdir, $module, $objectname, $newmask, $readdir = '
...
+	$error = 0;
+	if (empty($right)) {
+		$error++;
+	}
+	if ($action == 2 && empty($rightUpdated)) {
+		$error++;
+	}
+	if (!$error) {
+		// Open the file and read line by line
+		$handle = fopen($file, "r");
```
Comment (Hystepik) [inline htdocs/core/lib/modulebuilder.lib.php:444]: Here we prefer to use a file_get_contents() for security purpose

Comment (lamrani002) [reply htdocs/core/lib/modulebuilder.lib.php:444]: ok , i see

File: htdocs/core/lib/modulebuilder.lib.php [hunk 2]
```diff
@@ -416,3 +416,70 @@ function rebuildObjectSql($destdir, $module, $objectname, $newmask, $readdir = '
...
+					}
+					$lines[] = $line;
+					$i++;
+				}
+				break;
+			}
+			$i++;
+		}
+		$allContent = implode("", $lines);
+		if (str_contains($allContent, $right)) {
```
Comment (Hystepik) [inline htdocs/core/lib/modulebuilder.lib.php:467]: A better way to make this is to reforge entirely the /* BEGIN MODULEBUILDER PERMISSIONS / / END MODULEBUILDER PERMISSIONS */.
This way we can modify permission even if user as modified the file

Comment (lamrani002) [reply htdocs/core/lib/modulebuilder.lib.php]: I make some update if you can check and tell me if that what we need

**Comment by Hystepik** [comment]
@eldy seams ok to me

**Pull request merged**